File: fixtures/js/infinite_loop_no_exit/positive.ts
// MAINT010-infinite-loop-no-exit: positive fixture
// These loops should be flagged.

function spin(): void {
    let x = 0;
    while (true) {
        x++;
    }
}

function foreverFor(): void {
    let y = 0;
    for (;;) {
        y++;
    }
}

File: fixtures/rust/infinite_loop_no_exit/positive.rs
// MAINT010-infinite-loop-no-exit: positive fixture
// These loops should be flagged — they have no break, return, or panic.

fn spin() {
    let mut x = 0i32;
    loop {
        x += 1;
        let _ = x;
    }
}

fn call_in_loop() {
    loop {
        let _ = do_work();
    }
}

fn do_work() -> i32 {
    42
}
